pkgs/pub_semver/lib/src/version.dart

```diff
@@ -160,7 +166,7 @@ class Version implements VersionConstraint, VersionRange {
           // Return an integer part if possible, otherwise return the string
           // as-is
           int.tryParse(part) ?? part)
-      .toList();
+      .toList(growable: false);
```

Comment:
I don't think ungrowable and unmodifiable are the same thing. You can still modify entries, just not expand the list.